apps/server/src/services/integration-service/HttpIntegration.ts

```diff
@@ -45,18 +45,13 @@ export class HttpIntegration implements IIntegration<HttpSubscription> {
       }
 
       const parsedMessage = parseTemplateNested(message, state || {});
-      try {
-        const parsedUrl = new URL(parsedMessage);
-        this.emit(parsedUrl);
-      } catch (error) {
-        logger.error(LogOrigin.Tx, `HTTP Integration: ${error}`);
-      }
+      this.emit(parsedMessage);
     }
   }
 
-  async emit(path: URL) {
-    await got.get(path, {
-      retry: { limit: 0 },
+  emit(path: string) {
+    got.get(path, { retry: { limit: 0 } }).catch((err) => {
+      logger.error(LogOrigin.Tx, `HTTP Integration: ${err.code}`);
     });
```

Comment:
this seems ok to me, but since got is not handling anything, why would we not use node fetch instead?
the point of got would have been to simplify some operation (error handling, retrying, parsing...) but we seem to do none of that?

what do you think?

Reply:
I would like to add back the option for https at some point
and got handles the reusing of the sockets for us